File: php/src/Model/BulkTaskReport.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Outcome of a bulk create. A 200 does not mean everything was created: read `failed` and the per-entry results, whose `index` maps back to the input list.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class BulkTaskReport
{
    public function __construct(
        public readonly int $created,
        public readonly int $failed,
        /** @var list<BulkTaskResult> */
        public readonly array $results,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            created: (int) ($data['created'] ?? 0),
            failed: (int) ($data['failed'] ?? 0),
            results: Json::parseEach($data, 'results', [BulkTaskResult::class, 'fromArray']),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/BulkTaskReport.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * Outcome of a bulk create. A 200 does not mean everything was created: read {@code failed} and the per-entry results, whose {@code index} maps back to the input list.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class BulkTaskReport {
    private Integer created;
    private Integer failed;
    private List<BulkTaskResult> results;

    public Integer getCreated() { return created; }
    public Integer getFailed() { return failed; }
    public List<BulkTaskResult> getResults() { return results; }
}
